rs/p2p/quic_transport/src/connection_handle.rs

```diff
@@ -50,6 +54,64 @@ pub struct ConnectionHandle {
     conn_id: ConnId,
 }
 
+fn observe_conn_error(err: &ConnectionError, op: &str, metrics: &QuicTransportMetrics) {
+    match err {
+        // TODO: most likely this can be made infallible
+        ConnectionError::LocallyClosed => metrics
+            .connection_handle_errors_total
+            .with_label_values(&[op, ERROR_LOCALLY_CLOSED_CONN])
+            .inc(),
+        ConnectionError::ApplicationClosed(_) => metrics
+            .connection_handle_errors_total
+            .with_label_values(&[op, ERROR_APP_CLOSED_CONN])
+            .inc(),
```

Comment:
Yes, because the rest should be infallible. If I put explicit labels for the rest I think people may start ignoring them if they start to pop up. If a case is infallible and you see the counter going up, then people will automatically try to debug.